fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -821,6 +821,9 @@ default boolean recordExists(@Nonnull final Tuple primaryKey, @Nonnull final Iso
         final Tuple primaryKey = indexEntryPrimaryKey(index, entry.getKey());
         return loadRecordInternal(primaryKey, false).thenApply(record -> {
             if (record == null) {
+                if (getTimer() != null) {
+                    getTimer().increment(FDBStoreTimer.Counts.BAD_INDEX_ENTRY);
```

Comment:
This `if()` block should be moved under the `case ERROR` below.  The `SKIP` and `RETURN` options are used by index maintainers that wish to defer index maintenance until later, so those missing index entries are expected.

Reply:
Done.